woudc_data_registry/models.py

```diff
@@ -1163,12 +1250,113 @@ def sync(ctx):
         Station,
         Instrument,
         Deployment,
-        DataRecord
+        DataRecord,
     ]
+    contribution_models = list()
+
+    """function to form all the contributions from the other models lists"""
+    def build_contributions(contribution_models, project_models,
+                            station_models, instrument_models, country_models,
+                            deployment_models, contributor_models):
+
+        """turn lists holding essential information into dictionaries with key
+        that is common with another list in order to allow for fast lookup"""
+        country_dict = {country_models[i].country_id: country_models[i]
```

Comment:
Can do

```
blah_blah = { model.some_field: model for model in somethings  }
```

instead of using `range` for all these dictionaries, probably more readable that way.